pkg/graveler/graveler.go

```diff
@@ -1745,6 +1779,38 @@ func (g *KVGraveler) Commit(ctx context.Context, repositoryID RepositoryID, bran
 	return newCommitID, nil
 }
 
+func (g *KVGraveler) retryBranchUpdate(ctx context.Context, repositoryID RepositoryID, branchID BranchID, f BranchUpdateFunc) error {
+	const (
+		maxInterval = 5
+		setTries    = 3
+	)
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxInterval = maxInterval * time.Second
+
+	var try int
+	err := backoff.Retry(func() error {
+		// TODO(3586) - if the branch commit id hasn't changed, update the fields instead of fail
```

Comment:
https://github.com/treeverse/lakeFS/blob/b8bf1b28086358e960669de41a0e88af671584e3/design/accepted/metadata_kv/index.md?plain=1#L200

I know, but it will be assigned later.